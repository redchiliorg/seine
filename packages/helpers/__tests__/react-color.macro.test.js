const pluginTester = require('babel-plugin-tester');
const plugin = require('babel-plugin-macros');

pluginTester({
  plugin,
  babelOptions: { filename: __filename },
  snapshot: true,
  tests: [
    `
      // @flow
      import { SketchPicker } from '../react-color.macro';
      
      const DefaultPicker = SketchPicker;
      
      export default () => <DefaultPicker />;
      export const Picker = () => {
        const Picker = SketchPicker;
        return <Picker />;
      }
      export { SketchPicker };
    `,
  ],
});
