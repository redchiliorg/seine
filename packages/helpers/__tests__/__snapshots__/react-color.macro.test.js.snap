// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. macros: 1. macros 1`] = `
"
// @flow
import { SketchPicker } from '../react-color.macro';

const DefaultPicker = SketchPicker;

export default () => <DefaultPicker />;
export const Picker = () => {
  const Picker = SketchPicker;
  return <Picker />;
}
export { SketchPicker };

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

var _interopRequireDefault = require(\\"/home/mi6gan/projects/dodone/testbox/seine/node_modules/babel-preset-react-app/node_modules/@babel/runtime/helpers/interopRequireDefault\\");

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
Object.defineProperty(exports, \\"SketchPicker\\", {
  enumerable: true,
  get: function () {
    return _Sketch.default;
  }
});
exports.Picker = exports.default = void 0;

var _Sketch = _interopRequireDefault(require(\\"react-color/lib/Sketch\\"));

var _jsxFileName = \\"/home/mi6gan/projects/dodone/testbox/seine/packages/helpers/__tests__/react-color.macro.test.js\\";
const DefaultPicker = _Sketch.default;

var _default = () => React.createElement(DefaultPicker, {
  __source: {
    fileName: _jsxFileName,
    lineNumber: 6
  },
  __self: void 0
});

exports.default = _default;

const Picker = () => {
  const Picker = _Sketch.default;
  return React.createElement(Picker, {
    __source: {
      fileName: _jsxFileName,
      lineNumber: 9
    },
    __self: void 0
  });
};

exports.Picker = Picker;
"
`;
